### PR TITLE
[RFR] Formats

### PR DESCRIPTION
**Pull request by djhi**
[Trello card](https://trello.com/c/sxcxNlqd)

- [x] Add format property to field model
- [x] List format dynamically like transformers
- [x] Add display component based on format (type url)
- [x] Tests (to be continued)

### CODE REVIEW

File: src/app/js/dataset/DefaultColumn.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import fetchByUri from '../lib/fetchByUri';
+import { getToken } from '../user';
+import { field as fieldPropTypes } from '../propTypes';
+
+export class DatasetColumnComponent extends Component {
+    componentWillMount() {
+        const { column, resource } = this.props;
+        const linkTransformer = column.transformers && column.transformers.find(t => t.operation === 'LINK');
+        if (linkTransformer) {
+            const uri = resource[linkTransformer.args.find(a => a.name === 'reference').value];
```
Comment (ThieryMichel) [inline src/app/js/dataset/DefaultColumn.js:18]: it should not be the component that should make this call. But rather the saga that handle the change.

File: src/app/js/formats/uri/component.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import React, { PropTypes } from 'react';
+import { Link } from 'react-router';
+import { field as fieldPropTypes } from '../../propTypes';
+
+const UriComponent = ({ linkedResource, resource, field }) => {
```
Comment (ThieryMichel) [inline src/app/js/formats/uri/component.js]: i'll call it UriView, and don't forget the uppercase on the file name, all component start with one.

File: src/app/js/formats/uri/editionComponent.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import React, { Component, PropTypes } from 'react';
```
Comment (ThieryMichel) [inline src/app/js/formats/uri/editionComponent.js]: same for file name EditionComponent

**Review by ThieryMichel — CHANGES_REQUESTED** [review]

File: src/app/js/dataset/DefaultColumn.js (hunk 1 above)
Comment (djhi) [reply src/app/js/dataset/DefaultColumn.js:18]: what change ?

Comment (djhi) [reply src/app/js/dataset/DefaultColumn.js:18]: Imo, the only way to put it in saga is to (in `/dataset/sagas.js`)

- check for fields definitions and detect LINK
- loop over retrieved resources and call the fetchByUri for each linked column

Comment (djhi) [reply src/app/js/dataset/DefaultColumn.js:18]: Or to set up actions à la Twitter with normalized array of resources `{ data: { ... }, fetched: true|false }` etc. Seems overkill for now

Comment (ThieryMichel) [reply src/app/js/dataset/DefaultColumn.js:18]: No in updateField saga, when we change field transformer value, you get the change, and then check the uri.

Comment (djhi) [reply src/app/js/dataset/DefaultColumn.js:18]: we are on the public side here

Comment (ThieryMichel) [reply src/app/js/dataset/DefaultColumn.js:18]: my bad

**Review by ThieryMichel — APPROVED** [review]

**Pull request merged**